src/libsyntax/ext/base.rs

```diff
@@ -546,6 +557,7 @@ pub struct ExtCtxt<'a> {
     pub ecfg: expand::ExpansionConfig<'a>,
     pub crate_root: Option<&'static str>,
     pub feature_gated_cfgs: &'a mut Vec<GatedCfgAttr>,
+    pub loader: &'a mut MacroLoader,
```

Comment:
It's a sad thing that we have to use a trait and a trait object just for tests. Is there no other way? Can't we just create a dummy `macro_import::MacroLoader`?